include/libgit4cpp/Repository.h

```diff
@@ -233,8 +233,56 @@ class Repository
      * \param branch_name
     */
     bool branch_up_to_date(const std::string& branch_name);
+
 #endif
 
+
+    /**
+     * Create a new branch from the current branch.
+     * \param branch_name name of the new branch
+    */
+    LibGitReference new_branch(const std::string& branch_name);
+
+    /**
+     * Create a new branch from a specified existing branch.
+     * \param branch_name name of the new branch
+     * \param origin_branch_name name of the existing branch to checkout from
+     * \return The reference object of the new branch
+    */
+    LibGitReference new_branch(const std::string& branch_name, const std::string& origin_branch_name);
+
+    /**
+     * Returns the active branch in the repository.
+     * \return shorthand name of current branch (eg. master)
+    */
+    std::string get_current_branch();
+
+    /**
+     * List the longnames of all branches.
+     * \param type_flag defines which types of branches should be listed
+     *  0 -> All
+     *  1 -> Local
+     *  2 -> Remote
+     * \return vector of branch longnames
+    */
+    std::vector<std::string> list_branches(int type_flag = 0);
+
+    /**
+     * Checkout a branch with options.
```

Comment:
As far as I see, there are not so many options here, so maybe like this?
```suggestion
     * Check out selected files from a branch.
```